### ISSUE TITLE
Compile error due to Boost

### ISSUE THREAD

**Issue by farmboy0** [vcmi/vcmi#3351]
**Describe the bug**
Cmake finds all required boost components but the build still fails due to boost.
[cmake.log](https://github.com/vcmi/vcmi/files/13726453/cmake.log)
`cmake .. -DENABLE_ERM=ON -DENABLE_LAUNCHER=ON -DENABLE_PCH=ON -DENABLE_MONOLITHIC_INSTALL=OFF -DFORCE_BUNDLED_FL=OFF  -DBoost_DEBUG=ON`

Make will spit out:
$ make
-- -- -- Start of Git information
-- GIT_SHA1: 17b2ce2c81eeeb2ac1c46c38f963da2e8ec1d0e4
-- -- -- End of Git information
[  0%] Built target update_version
lib/CMakeFiles/vcmi.dir/build.make:4381: *** target pattern contains no '%'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:470: lib/CMakeFiles/vcmi.dir/all] Error 2
make: *** [Makefile:156: all] Error 2

With ninja you get a better error message:
$ ninja
ninja: error: 'Boost::thread-NOTFOUND', needed by 'bin/libvcmi.so', missing and no known rule to make it

$ cmake --version
cmake version 3.27.7

**Expected behavior**
Build will succeed.

**Actual behavior**
Build error due to boost.

**Did it work earlier?**
Yes, it worked earlier, boost version was the same so probnably cmake changed but i dont have the build log anymore.

**Screenshots**

**Version**
 - OS: Gentoo Linux
 - Version: 1.4.1 but the error will happen on all versions I tried (1.2.1, 1.3.2)

**Additional context**
Its something that changed on my system, as I could build 1.2.1 successfully in July but Im not familiar enough with cmake or Boost to find the problem. Help is appreciated.

**Comment by kambala-decapitator** [comment]
add `--debug-find` to your cmake invocation, it will spit huge log with info how/where it tries to find dependencies

**Comment by kambala-decapitator** [comment]
don't see anything obvious there, only this is suspicious:

  Policy CMP0111 is not set: An imported target missing its location property
  fails during generation.  Run "cmake --help-policy CMP0111" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  IMPORTED_LOCATION not set for imported target "Boost::thread" configuration
  "RelWithDebInfo"

**Comment by kambala-decapitator** [comment]
also check your CMakeCache, I believe there should be something with NOTFOUND

**Comment by farmboy0** [comment]
There is nothing with NOTFOUND in CMakeCache
[CMakeCache.txt](https://github.com/vcmi/vcmi/files/13730108/CMakeCache.txt)

**Comment by kambala-decapitator** [comment]
sorry, no idea then. Try reinstalling boost and/or configuring VCMI in another build folder.

**Comment by 1aNam** [comment]
Hi. Setting `Boost_NO_BOOST_CMAKE=ON` in VCMI ebuild did the right job. [This property overrides CMake's FindBoost module](https://cmake.org/cmake/help/latest/module/FindBoost.html#boost-cmake) that cannot for some reason pass found Boost libraries to Ninja for further linking.

I'm running Gentoo with dev-util/cmake-3.28.1, dev-util/cmake-3.28.1. Compiled VCMI 1.4.2 seems to run fine now for me.

**Comment by farmboy0** [comment]
Thank you @1aNam that worked.

**Comment by DJManas** [comment]
> `Boost_NO_BOOST_CMAKE=ON`

Great, thanks. Edited ebuild in PG_Overlay with this and it builds.

**Comment by msva** [comment]
I also fixed ebuild in gamerlay and bumoed it to 1.4.2 as well.
Thanks for hack with boost redefinition

**Comment by w345731923** [comment]
Alternatively, using -DBoost_NO_BOOST_CMAKE=BOOL:ON in CMake is also an option.

**Comment by sanfanling** [comment]
> I also fixed ebuild in gamerlay and bumoed it to 1.4.2 as well. Thanks for hack with boost redefinition

Hi, how to get the latest version ebuild? Thx

**Comment by DJManas** [comment]
> > I also fixed ebuild in gamerlay and bumoed it to 1.4.2 as well. Thanks for hack with boost redefinition
> 
> Hi, how to get the latest version ebuild? Thx

```
sudo eselect repository enable gamerlay
sudo emaint sync -r gamerlay
sudo emerge -av vcmi
```

eselect repository enable -> enables repository gamerlay, where vcmi 1.4.2 is located
emaint sync -> syncs repo to local computer
emerge -> installs vcmi 1.4.2

For version 9999 I can provide you my ebuild, but then you should setup your local portage overlay, its possible, but a little more setup is needed

**Comment by sanfanling** [comment]
> > > I also fixed ebuild in gamerlay and bumoed it to 1.4.2 as well. Thanks for hack with boost redefinition
> > 
> > 
> > Hi, how to get the latest version ebuild? Thx
> 
> ```
> sudo eselect repository enable gamerlay
> sudo emaint sync -r gamerlay
> sudo emerge -av vcmi
> ```
> 
> eselect repository enable -> enables repository gamerlay, where vcmi 1.4.2 is located emaint sync -> syncs repo to local computer emerge -> installs vcmi 1.4.2
> 
> Maybee you need to put: _games-strategy/vcmi ~amd64_
> 
> To _/etc/portage/package.acceptkey_words_ file or subfile in that directory.
> 
> For version 9999 I can provide you my ebuild, but then you should setup your local portage overlay, its possible, but a little more setup is needed.

Hi, I have gamerlay, and the version in this overlay is still 1.4.2, and seems no maintainers.
It's best if you could provide your ebuild, I can handle this. Thank you very much  
my email is : xujia19@outlook.com

**Comment by DJManas** [comment]
I will add it here, maybee someone else might get use of it.

[vcmi-9999.zip](https://github.com/user-attachments/files/17009644/vcmi-9999.zip)

**Comment by sanfanling** [comment]
> I will add it here, maybee someone else might get use of it.
> 
> [vcmi-9999.zip](https://github.com/user-attachments/files/17009644/vcmi-9999.zip)

WOW, it's daily version, awesome!!! It's compiling. Can I share it in gentoo forums? I wrote a post there seeking for the ebuild.

**Comment by sanfanling** [comment]
> I will add it here, maybee someone else might get use of it.
> 
> [vcmi-9999.zip](https://github.com/user-attachments/files/17009644/vcmi-9999.zip)

Oh NO....

-- Installing: /var/tmp/portage/games-strategy/vcmi-9999/image/usr/bin/vcmiclient
-- Set non-toolchain portion of runtime path of "/var/tmp/portage/games-strategy/vcmi-9999/image/usr/bin/vcmiclient" to "/usr/lib64/vcmi"
CMake Error at clientapp/cmake_install.cmake:70 (file):
  file INSTALL cannot find
  "/var/tmp/portage/games-strategy/vcmi-9999/work/vcmi-9999/clientapp/icons/vcmiclient.16x16.png":
  No such file or directory.
Call Stack (most recent call first):
  cmake_install.cmake:77 (include)


FAILED: CMakeFiles/install.util 
cd /var/tmp/portage/games-strategy/vcmi-9999/work/vcmi-9999_build && /usr/bin/cmake -P cmake_install.cmake
ninja: build stopped: subcommand failed.

Any hits?

**Comment by sanfanling** [comment]
> Probably vcmi bug: #4594
> 
> Let me check that PR...



> Probably vcmi bug: #4594
> 
> Let me check that PR...

Bingo, after re-gitting && re-compiling. Thank you guys.

**Comment by DJManas** [comment]
> > I will add it here, maybee someone else might get use of it.
> > [vcmi-9999.zip](https://github.com/user-attachments/files/17009644/vcmi-9999.zip)
> 
> WOW, it's daily version, awesome!!! It's compiling. Can I share it in gentoo forums? I wrote a post there seeking for the ebuild.

Sure. I had tried to add it to [PG_Overlay](https://gitlab.com/Perfect_Gentleman/PG_Overlay/-/merge_requests/3), but the merge request is still hanging. If it does someone happy, feel free to share. I was about to do my own overlay, where I would keep these things, but well. Not enough time to check what I need to fulfill to have my own overlay or what I need to post to original gentoo ebuild repository.

Would be nice If I implemented the patches as well.
